src/config/searchWorker.ts

```diff
@@ -130,16 +182,67 @@ export default () => {
         list,
         sortBy,
         sortOrder,
+        nodeListingFilters: { isNodeListing, searchParams },
     }: {
         searchText: string
         list: unknown[]
         sortBy: string
         sortOrder: SortingOrder
+        nodeListingFilters: {
+            isNodeListing: boolean
+            searchParams: Record<string, string>
+        }
     }) => {
         const searchTextLowerCased = searchText.trim().toLowerCase()
         let filteredList = [...list]
 
-        if (searchTextLowerCased !== '' && list?.length) {
+        if (isNodeListing) {
+            filteredList = list.filter((item: Record<string, any>) => {
+                const isK8sVersionFilterAppliedAndMatchFound =
+                    !searchParams[NODE_K8S_VERSION_KEY] ||
+                    item[NODE_K8S_VERSION_KEY] === searchParams[NODE_K8S_VERSION_KEY]
+
+                const doesAnyNodeSearchKeyExists = Object.values(NODE_SEARCH_KEYS).some((key) =>
```

Comment:
Suggestions:
- Can early return if isK8sVersionFilterAppliedAndMatchFound is false to save iterations
- Can combine both operations of doesAnyNodeSearchKeyExists and doesItemHaveAnyMatchingSearchKey using Object.entrties.filter then every (Can discuss)